src/uniforms/sampler.rs

```diff
@@ -140,6 +140,36 @@ impl ToGlEnum for DepthTextureComparison {
     }
 }
 
+
+/// The border color to use for `SamplerWrapFunction::BorderClamp`.
+///
+/// This type encapsulates `[f32; 4]` internally as `[u32; 4]` so that
+/// it implements Hash and Eq. The conversions to and from `[f32; 4]`
+/// are provided by the `From` trait.
+#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq)]
+pub struct BorderColor([u32; 4]);
+
+impl From<[f32; 4]> for BorderColor {
+    #[inline]
+    fn from(value: [f32; 4]) -> BorderColor {
+        BorderColor(unsafe { std::mem::transmute(value) })
```

Comment:
there is no need to use unsafe here: you can use array.map with `f32::to_bits`. in the end it will most likely compile to the same code.